### PR TITLE
Use more signatures - cv/basetest/autotest

### PR DESCRIPTION
**Pull request by okurz**
* Use proper perl "signatures" in common files - autotest
* Use proper perl "signatures" in common files - basetest
* Use proper perl "signatures" in common files - cv

### CODE REVIEW

File: autotest.pm [hunk 1]
```diff
@@ -262,9 +254,7 @@ sub handle_sigterm {
     _exit(1);
 }
 
-sub start_process {
-    my $child;
-
+sub start_process ($child = undef) {
```
Comment (perlpunk) [inline autotest.pm:257]: Is it intended that `$child` can be passed as an argument now?

File: basetest.pm [hunk 2]
```diff
@@ -453,11 +418,7 @@ sub record_resultfile {
     $self->write_resultfile($filename, $output);
 }
 
-sub record_serialresult {
-    my ($self, $ref, $res, $string) = @_;
-
-    $string //= '';
-
+sub record_serialresult ($self, $ref, $res, $string = '') {
```
Comment (perlpunk) [inline basetest.pm:421]: `$string` might end up as `undef`

File: basetest.pm [hunk 3]
```diff
@@ -499,10 +456,7 @@ test details and returns it.
 
 =cut
 
-sub record_testresult {
-    my ($self, $result, %args) = @_;
-    $result //= 'unk';
-
+sub record_testresult ($self, $result = 'unk', %args) {
```
Comment (perlpunk) [inline basetest.pm:459]: `$result` might end up as undef

File: basetest.pm [hunk 4]
```diff
@@ -559,10 +511,7 @@ add screenshot with 'unk' result if an image is available
 
 =cut
 
-sub take_screenshot {
-    my ($self, $res) = @_;
-    $res ||= 'unk';
-
+sub take_screenshot ($self, $res = 'unk') {
```
Comment (perlpunk) [inline basetest.pm:514]: same here

File: autotest.pm (hunk 1 above)
Comment (okurz) [reply autotest.pm]: No, that does not make sense. Thanks.

**Review by okurz — COMMENTED** [review]
updated

**Review by Martchus — CHANGES_REQUESTED** [review]

File: basetest.pm [hunk 5]
```diff
@@ -499,10 +457,8 @@ test details and returns it.
 
 =cut
 
-sub record_testresult {
-    my ($self, $result, %args) = @_;
+sub record_testresult ($self, $result = undef, %args) {
```
[the remark this answers is not in the document]
Comment (okurz) [reply basetest.pm:460]: yes because `%args` means "[0;inf["

File: basetest.pm [hunk 6]
```diff
@@ -24,9 +24,7 @@ sub import {
     warnings->unimport('experimental::signatures');
 }
 
-sub new {
-    my ($class, $category) = @_;
-    $category ||= 'unknown';
+sub new ($class, $category = 'unknown') {
```
[the remark this answers is not in the document]
Comment (okurz) [reply basetest.pm]: sorry, you just need to mention it … 27 more times and I will get it :)

**Review by Martchus — APPROVED** [review]

File: basetest.pm (hunk 5 above)
[the remark this answers is not in the document]
Comment (perlpunk) [reply basetest.pm:461]: Right. An array or hash is like a catchall.

**Pull request merged**